lctrie_ip.c

```diff
@@ -79,13 +79,9 @@ size_t subnet_dedup(lct_subnet_t *subnets, size_t size) {
       // assume that the prior defined subnet is the desired one,
       // dis-allowing redefinition of that subnet elsewhere,
       // ex. bogon file, BGP ASN list, user specified subnets
-      //
-      // slide the rest of the array over the second value.  if we're at the
-      // end of the array, just let it drop off.
-      if ((j + 1) < size)
-        memmove(&subnets[j], &subnets[j + 1], (size - (j + 1)) * sizeof(lct_subnet_t));
-      --size;
```

Comment:
You removed the array size decrement and the memmove(), so I fail to see how this change _ever_ removes duplicates from the data set.

Reply:
Even if the array size decrement is left in, size was passed by value and so the actual value will not be changed outside of the function.

In lctrie_test.c (https://github.com/chuckination/lctrie/blob/master/lctrie_test.c#L229-L230)
Size is decremented by the number of duplicates and the array is re-alloced -- this removes the duplicates since they will all be clustered in the last (ndup) elements